main.go

```diff
@@ -16,29 +16,356 @@
+// certificate, and extracts the repository's workflow file to ensure its legitimacy.
+func verifySignature(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "error reading http request body", http.StatusInternalServerError)
+		fmt.Println(err)
+		return
+	}
+
+	// Get most recent Rekor entry uuid.
+	rekorClient, err := rekor.NewClient(options.DefaultRekorURL)
+	if err != nil {
+		http.Error(w, "error initializing Rekor client", http.StatusInternalServerError)
+		fmt.Println(err)
+		return
+	}
+	uuids, err := cosign.FindTLogEntriesByPayload(ctx, rekorClient, reqBody)
+	if err != nil || len(uuids) == 0 {
+		http.Error(w, "error fetching tlog entries", http.StatusInternalServerError)
+		fmt.Println(err)
+		return
+	}
+	uuid := uuids[len(uuids)-1] // ignore past entries.
+
+	// Verify tlog entry to make sure it is actually in the log.
+	entry, err := verifyTLogEntry(ctx, rekorClient, uuid)
+	if err != nil {
+		http.Error(w, "error verifying tlog entry", http.StatusInternalServerError)
+		fmt.Println(err)
+		return
+	}
+
+	// Extract certificate and get repo reference & path.
+	certs, err := extractCerts(entry)
+	if err != nil || len(certs) == 0 {
+		http.Error(w, "error extracting certificate from entry", http.StatusInternalServerError)
+		fmt.Println(err)
+		return
+	}
+	if len(certs) > 1 {
+		http.Error(w, "multiple certificates found for the entry", http.StatusInternalServerError)
+		return
+	}
+	cert := certs[0]
+	var repoRef string
+	var repoPath string
+	for _, ext := range cert.Extensions {
+		// OID source: https://github.com/sigstore/fulcio/blob/96ef49cc7662912ba37d46f738757e8d8d5b5355/docs/oid-info.md#L33
+		// TODO: retrieve these by name.
+		if ext.Id.String() == "1.3.6.1.4.1.57264.1.6" {
+			repoRef = string(ext.Value)
+		}
+		if ext.Id.String() == "1.3.6.1.4.1.57264.1.5" {
+			repoPath = string(ext.Value)
+		}
+	}
+
+	if len(repoRef) == 0 || len(repoPath) == 0 {
+		http.Error(w, "error extracting repo ref or path from certificate", http.StatusInternalServerError)
+		return
+	}
+
+	// Split repo path into owner and repo name.
+	ownerName := repoPath[:strings.Index(repoPath, "/")]
+	repoName := repoPath[strings.Index(repoPath, "/")+1:]
+
+	// Verify that the repository and branch of the cert and request are equal.
+	reqRepo := r.Header["Repository"]
+	reqBranch := r.Header["Branch"]
+	if len(reqRepo) == 0 || len(reqBranch) == 0 || reqRepo[0] != repoPath || reqBranch[0] != repoRef {
+		http.Error(w, "repository and branch of cert doesn't match that of request", http.StatusInternalServerError)
+		return
+	}
+
+	// Get workflow file from repo reference.
+	// TODO: use GITHUB_TOKEN from workflow to make the api call.
+	client := github.NewClient(nil)
+	opts := &github.RepositoryContentGetOptions{Ref: repoRef}
+	contents, _, _, err := client.Repositories.GetContents(ctx, ownerName, repoName, ".github/workflows/scorecards.yml", opts)
```

Comment:
This assumes everyone is storing scorecard as `scorecards.yml`. This might not be the case.

Reply:
Isn't this the default naming convention used by the template? The cert doesn't seem to store the workflows name, only [these values](https://github.com/sigstore/fulcio/blob/96ef49cc7662912ba37d46f738757e8d8d5b5355/docs/oid-info.md#L33), so I'm not sure how to retrieve this.

Reply:
@laurentsimon fixed this to just iterate thru all the workflows in the given repo and select the one with the correct name